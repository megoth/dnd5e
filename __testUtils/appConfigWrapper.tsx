import React, { ReactNode } from "react";
import AppConfigProvider from "../src/contexts/appConfig";
import mockAppConfig from "./mockAppConfig";

interface Props {
  children: ReactNode;
}

export default function AppConfigWrapper({ children }: Props) {
  const { errorsUrl, translationsUrl } = mockAppConfig();
  return (
    <AppConfigProvider errorsUrl={errorsUrl} translationsUrl={translationsUrl}>
      {children}
    </AppConfigProvider>
  );
}
