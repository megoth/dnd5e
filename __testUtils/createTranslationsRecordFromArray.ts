export default function createTranslationsRecordFromArray(
  url,
  list: string[]
): Record<string, string> {
  return list.reduce((memo: Record<string, string>, key) => {
    // eslint-disable-next-line no-param-reassign
    memo[`${url}#${key}`] = key;
    return memo;
  }, {});
}
