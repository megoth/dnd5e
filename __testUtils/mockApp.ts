import { AppModel, createApp, getBundleKey } from "../src/models/app";
import mockFluentBundle from "./mockFluentBundle";
import { appVocabURL } from "./mockAppIndexDataset";
import mockLanguage, { defaultLocale } from "./mockLanguage";
import mockResourceBundleMap from "./mockResourceBundleMap";
import mockAppIndex from "./mockAppIndex";

export const errorsURL = "https://example.com/global/errors.ttl";
export const faqsURL = "https://example.com/global/faqs.ttl";
export const translationsURL = "https://example.com/global/translations.ttl";
export const localizationsURL =
  "https://example.com/global/translations.en-US.ttl";

export function mockAppPropValue(val, bundleName = "global", locale = "en-US") {
  return bundleName
    ? {
        [getBundleKey(locale, bundleName)]: val,
      }
    : {
        [locale]: val,
      };
}

export default function mockApp(overrides: Partial<AppModel> = {}): AppModel {
  const currentLocale = overrides.currentLocale || defaultLocale;
  return {
    appVocabURL: overrides.appVocabURL || appVocabURL,
    currentLocale,
    fluentBundles: {
      [currentLocale]: mockFluentBundle({}, translationsURL, {
        locale: currentLocale,
      }),
      ...overrides.fluentBundles,
    },
    languages: [mockLanguage(currentLocale), ...(overrides.languages || [])],
    resourceBundles: {
      ...mockResourceBundleMap({
        label: "admin",
        locale: currentLocale,
        data: {
          errors: null,
          faqs: null,
          localizations: null,
          translations: null,
        },
      }),
      ...mockResourceBundleMap({
        locale: currentLocale,
        data: {
          errors: null,
          faqs: null,
          localizations: null,
          translations: null,
        },
      }),
      ...overrides.resourceBundles,
    },
  };
}

export function mockNorwegianApp(overrides: Partial<AppModel> = {}): AppModel {
  const locale = "nb-NO";
  return mockApp({
    currentLocale: locale,
    fluentBundles: {
      [locale]: mockFluentBundle({}, translationsURL, {
        locale,
      }),
    },
    languages: [
      mockLanguage(locale, {
        languageCode: "ðŸ‡³ðŸ‡´",
      }),
    ],
    resourceBundles: {
      ...mockResourceBundleMap({
        locale,
        data: {
          errors: null,
          faqs: null,
          localizations: null,
          translations: null,
        },
      }),
    },
    ...overrides,
  });
}

export function mockEmptyApp(): AppModel {
  return createApp(defaultLocale, appVocabURL, null);
}

export function mockUnloadedApp(): AppModel {
  return createApp(defaultLocale, appVocabURL, mockAppIndex());
}
