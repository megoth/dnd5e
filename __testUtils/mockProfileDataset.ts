import {
  addStringNoLocale,
  addUrl,
  mockSolidDatasetFrom,
  mockThingFrom,
  setThing,
} from "@inrupt/solid-client";
import { vcard, foaf, rdf } from "rdf-namespaces";
import { chain } from "../src/utils";

export const aliceWebIdDatasetUrl = "http://example.com/alice";
export const aliceWebIdUrl = `${aliceWebIdDatasetUrl}#me`;
export const aliceName = "Alice";
export const aliceNick = "A";
export const alicePhoto = "http://example.com/alice.jpg";

export function mockProfileDataset(webId = aliceWebIdUrl) {
  const profile = chain(
    mockThingFrom(webId),
    (t) => addStringNoLocale(t, vcard.fn, aliceName),
    (t) => addStringNoLocale(t, vcard.nickname, aliceNick),
    (t) => addUrl(t, vcard.hasPhoto, alicePhoto),
    (t) => addUrl(t, rdf.type, foaf.Person),
    (t) => addUrl(t, foaf.openid, aliceWebIdUrl)
  );
  return chain(mockSolidDatasetFrom(aliceWebIdDatasetUrl), (d) =>
    setThing(d, profile)
  );
}
