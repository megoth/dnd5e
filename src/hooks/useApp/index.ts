import { useContext, useEffect } from "react";
import { AppContext } from "../../contexts/app";
import { appIsLoadingLocalizations } from "../../models/app";

export default function useApp(bundles: string[] = []) {
  const { app, setBundles } = useContext(AppContext);
  useEffect(() => {
    if (!app || !appIsLoadingLocalizations(app, bundles)) {
      return;
    }
    setBundles(bundles);
  }, [app, bundles, setBundles]);
  return app;
}
