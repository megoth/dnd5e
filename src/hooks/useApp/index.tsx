import { useContext } from "react";
import AppContext from "./context";

export default function useApp() {
  return useContext(AppContext);
}
