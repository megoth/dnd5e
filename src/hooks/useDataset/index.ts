import useSWR from "swr";
import { getSolidDataset } from "@inrupt/solid-client";
import { useSession } from "@inrupt/solid-ui-react";

export default function useDataset(url) {
  const { fetch } = useSession();
  return useSWR([url, "dataset"], () => getSolidDataset(url, { fetch }));
}
