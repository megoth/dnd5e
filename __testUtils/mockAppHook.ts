import mockApp from "./mockApp";

export default function mockAppHook(mock: jest.Mock, app = mockApp()) {
  mock.mockReturnValue(app);
}
