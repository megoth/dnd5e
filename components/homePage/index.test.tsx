import React from "react";
import { render } from "@testing-library/react";
import useApp from "../../src/hooks/useApp";
import mockAppHook from "../../__testUtils/mockAppHook";
import HomePage from "./index";
import useDataset from "../../src/hooks/useDataset";
import { mockProfileDataset } from "../../__testUtils/mockProfileDataset";

jest.mock("../../src/hooks/useApp");
const mockedAppHook = useApp as jest.Mock;

jest.mock("../../src/hooks/useDataset");
const mockedUseDataset = useDataset as jest.Mock;

describe("HomePage", () => {
  it("renders", () => {
    mockAppHook(mockedAppHook);
    mockedUseDataset.mockReturnValue(mockProfileDataset());
    const { asFragment } = render(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
