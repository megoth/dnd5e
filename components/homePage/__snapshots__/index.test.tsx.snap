// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders 1`] = `
<DocumentFragment>
  <main>
    <section
      class="hero"
    >
      <div
        class="h-screen text-white flex flex-col justify-center items-center leading-normal px-4 bg-gradient-to-t from-black relative text-center"
      >
        <picture>
          <source
            media="(min-width: 640px)"
            srcset="/images/appIcon/icons8-dungeons-and-dragons-128.png"
          />
          <source
            media="(min-width: 480px)"
            srcset="/images/appIcon/icons8-dungeons-and-dragons-96.png"
          />
          <img
            alt="[Translation for https://example.com/global/translations.ttl#appLogoDescription does not exist]"
            src="/images/appIcon/icons8-dungeons-and-dragons-64.png"
          />
        </picture>
        <h1
          class="text-3xl my-1 font-serif"
        >
          [Translation for https://example.com/global/translations.ttl#appName does not exist]
        </h1>
        <p
          class="my-1 max-w-xl"
        >
          [Translation for https://example.com/global/translations.ttl#appPitch does not exist]
        </p>
        <a
          class="my-1 max-w-3xl text-red-600 underline focus:outline-none focus:ring-2 focus:ring-red-600"
          href="/about"
        >
          [Translation for https://example.com/global/translations.ttl#learnMore does not exist]
        </a>
        <p
          class="text-2xl my-1"
        >
          [Translation for https://example.com/global/translations.ttl#workInProgress does not exist]
        </p>
        <p
          class="text-xs text-gray-600 absolute bottom-1 right-2"
        >
          Photo by 
          <a
            class="underline focus:outline-none focus:ring-2 focus:ring-gray-600"
            href="https://unsplash.com/@steve_j?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
          >
            Steve Johnson
          </a>
           on 
          <a
            class="underline focus:outline-none focus:ring-2 focus:ring-gray-600"
            href="https://unsplash.com/s/photos/pen-and-paper?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
          >
            Unsplash
          </a>
        </p>
      </div>
    </section>
  </main>
  <footer
    class="container mx-auto px-3 md:px-0 md:flex mt-2 max-w-3xl"
  >
    <div
      class="md:flex-grow mb-2"
    >
      <a
        class="text-red-600 underline mr-1 p-1 focus:outline-none focus:ring-2 focus:ring-red-600"
        href="/about"
      >
        [Translation for https://example.com/global/translations.ttl#aboutTitle does not exist]
      </a>
      <a
        class="text-red-600 underline mr-1 p-1 focus:outline-none focus:ring-2 focus:ring-red-600"
        href="/faq"
      >
        [Translation for https://example.com/global/translations.ttl#faqShort does not exist]
      </a>
    </div>
    <div
      class="mb-2"
    />
  </footer>
</DocumentFragment>
`;
